fearless/Common/Services/ExtrinsicService/ExtrinsicService.swift

```diff
@@ -13,10 +13,24 @@ protocol ExtrinsicOperationFactoryProtocol {
 
     func submit(
         _ closure: @escaping ExtrinsicBuilderClosure,
-        signer: SigningWrapperProtocol
+        signer: SigningWrapperProtocol,
+        nonceShift: UInt32
```

Comment:
The drawback of this approach is that nonce still requested for each extrinsic. Moreover it might be a problem because there is no guarantee that it will not change somewhere in the middle while a list of transaction is being submitted one by one. So I suggest to have a method that allows to submit multiple extrinsic at once. This allows to request nonce ones and increment it for each extrinsic . The simplest solution that comes to mind is to have a parameter numberOfExtrinsics. Then closure parameter type should be also changed and include index of extrinsic for which builder should be setup.